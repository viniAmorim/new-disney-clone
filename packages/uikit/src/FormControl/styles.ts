import styled from 'styled-components';

export type StyledFormControlProps = {
  isValid: boolean;
};

export const FormControl = styled.div`
  display: flex;
  flex-direction: column;
  margin-bottom: 8px;
`;

export const FormControlLabel = styled.label<StyledFormControlProps>`
  color: ${({ theme }) => theme.colors['blue-100']};
  height: 24px;
  margin: 6px 0px;
  width: 229px;
  ${({ theme }) => theme.typography['subheader-heavy']};
`;

export const FormControlError = styled.label`
  color: ${({ theme }) => theme.colors.error};
  margin-top: 6px;
`;
