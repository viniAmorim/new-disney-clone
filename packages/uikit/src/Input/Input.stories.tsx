import React from 'react';

import { Story, Meta } from '@storybook/react';

import Input, { InputProps } from '~/Input';

export default {
  title: 'Form/Input',
  component: Input,
} as Meta;

const Template: Story<InputProps> = (args) => <Input {...args} />;

export const Default = Template.bind({});
Default.args = {
  placeholder: 'Text content for input',
};

export const Error = Template.bind({});
Error.args = {
  placeholder: 'Text content for input',
  isValid: false,
};

export const Mask = Template.bind({});
Mask.args = {
  placeholder: 'Text content for input',
  format: '(##) ####-####',
  mask: '_',
};
