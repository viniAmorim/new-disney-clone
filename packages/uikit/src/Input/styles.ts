import { NumberFormatProps } from 'react-number-format';

// semantic error TS4023: Exported variable
import NumberFormat from '../../../../node_modules/react-number-format/dist/react-number-format';

import styled, { css } from 'styled-components';

export interface StyledInputProps extends NumberFormatProps {
  isValid?: boolean;
}

const variantModifier = {
  default: css`
    border: 1px solid ${({ theme }) => theme.colors['gray-400']};
  `,
  success: css`
    border: 1px solid ${({ theme }) => theme.colors.success};
  `,
  error: css`
    border: 1px solid ${({ theme }) => theme.colors.error};
  `,
};

const InputCss = css`
  background: ${({ theme }) => theme.colors.white};
  border: 1px solid ${({ theme }) => theme.colors['gray-400']};
  border-radius: 4px;
  padding: 12px 16px;

  &:focus,
  &:hover {
    border: 1px solid ${({ theme }) => theme.colors['blue-300']};
    outline: none;

    ::placeholder {
      color: ${({ theme }) => theme.colors['gray-100']};
    }
  }
`;

export const Input = styled(NumberFormat).withConfig({
  shouldForwardProp: (prop, defaultValidatorFn) => {
    return !['isValid'].includes(prop as string) && defaultValidatorFn(prop);
  },
})<StyledInputProps>`
  ${InputCss}

  ${({ isValid }) =>
    isValid ? variantModifier.default : variantModifier.error};
`;

export const RegularInput = styled.input.withConfig({
  shouldForwardProp: (prop, defaultValidatorFn) =>
    !['isValid'].includes(prop) && defaultValidatorFn(prop),
})<StyledInputProps>`
  ${InputCss}

  ${({ isValid }) =>
    isValid ? variantModifier.default : variantModifier.error};
`;

export const InputError = styled.text`
  margin-top: 6px;
  color: ${({ theme }) => theme.colors.error};
`;
