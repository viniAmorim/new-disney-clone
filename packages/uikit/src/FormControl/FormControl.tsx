import React, { BaseHTMLAttributes, cloneElement, ReactElement } from 'react';

import * as Styled from './styles';

export interface FormControlProps
  extends BaseHTMLAttributes<HTMLDivElement>,
    Styled.StyledFormControlProps {
  /**
   * Error
   */
  error?: string;
  /**
   * Label
   */
  label: string;
  /**
   * FormControl content
   */
  children: ReactElement;
}

const FormControl = ({ error, label, children, ...rest }: FormControlProps) => {
  const Component = cloneElement(children, {
    isValid: error === undefined,
  });
  return (
    <Styled.FormControl {...rest}>
      {label && (
        <Styled.FormControlLabel isValid={error === undefined}>
          {label}
        </Styled.FormControlLabel>
      )}

      {Component}

      {error && <Styled.FormControlError>{error}</Styled.FormControlError>}
    </Styled.FormControl>
  );
};

export default FormControl;
