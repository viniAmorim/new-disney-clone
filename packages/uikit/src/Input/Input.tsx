import React, { forwardRef } from 'react';

import * as Styled from './styles';

export interface InputProps extends Styled.StyledInputProps {
  /**
   * Placeholder
   */
  placeholder: string;
}

const Input = forwardRef(function({ isValid, mask, ...rest }: InputProps, ref) {
  if (mask) {
    return <Styled.Input ref={ref} isValid={isValid} {...rest} />;
  }

  return <Styled.RegularInput ref={ref} isValid={isValid} {...rest} />;
});

export default Input;
