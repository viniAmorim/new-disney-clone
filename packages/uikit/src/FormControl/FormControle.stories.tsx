import React from 'react';

import { Story, Meta } from '@storybook/react';

import FormControl, { FormControlProps } from '~/FormControl';
import Input from '~/Input';

export default {
  title: 'Form/FormControl',
  component: FormControl,
} as Meta;

const Template: Story<FormControlProps> = (args) => <FormControl {...args} />;

export const Default = Template.bind({});
Default.args = {
  label: 'Label Input text default:',
  children: <Input placeholder="Text content for input" />,
};

export const Error = Template.bind({});
Error.args = {
  label: 'Label Input text default:',
  children: <Input isValid={false} placeholder="Text content for input" />,
  error: 'Campo obrigat√≥rio',
};
