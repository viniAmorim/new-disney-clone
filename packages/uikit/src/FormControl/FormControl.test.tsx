/* eslint-disable @typescript-eslint/no-var-requires */
import React from 'react';

import { render } from '~/test/utils';

import Input from '../Input';
import FormControl from './FormControl';

describe('Input', () => {
  it('Should render the component', () => {
    const { getByRole } = render(
      <FormControl label="Label input text default" isValid>
        <Input placeholder="Text content for input" />
      </FormControl>,
    );

    const FormControlElement = getByRole('textbox');

    expect(FormControlElement).toBeInTheDocument();
    expect(FormControlElement).toMatchSnapshot();
  });
});
