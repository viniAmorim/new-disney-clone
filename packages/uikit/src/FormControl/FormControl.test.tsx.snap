// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Should render the component 1`] = `
<input
  class="sc-gsTEea entQzz"
  placeholder="Text content for input"
/>
`;
